### PR TITLE
New Quirk: Flutter

### PR DESCRIPTION
**Pull request by ShamanSliph**
# About The Pull Request

Adds a new Quirk to the game Called Flutter. Letting players move while in a Pressurized Low Gravity Environment. I went for a quirk instead of a specific Specie thing, so people could easily add it to character who it might fit, but are not that specific species. Like a Human who is clearly meant to be a demon, or a Lizard with Dragon wings.

The point value of 1 might be to cheap. But its so uncommon I'm not sure it would be right to make it worth 2.
PS: First time contributing to a project. Sorry if this was done incorrectly.

Changes:
Defined TRAIT_FLUTTER in code/__SPLURTCODE/DEFINES/traits.dm
Added Quirk to modular_splurt/code/datums/traits/good.dm
Added new File modular_splurt/code/modules/mob/mob_movement.dm

## Why It's Good For The Game

It was a request by Butters. Plus I wanted to dip my toes into programming in byond. So this seemed like a good place to start. 

## A Port?

While the idea was taken from TG's Moth code, the implementation is different.

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [X] You documented all of your changes.

## Changelog

:cl:
add:  Added a new quirk: Flutter
/:cl:

### CODE REVIEW

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -215,6 +215,10 @@
 /mob/Process_Spacemove(movement_dir = 0)
 	if(HAS_TRAIT(src, TRAIT_SPACEWALK) || spacewalk || ..())
 		return TRUE
+	var/turf/T = get_turf(src)
+	var/datum/gas_mixture/environment = T.return_air()
+	if(HAS_TRAIT(src, TRAIT_FLUTTER)&& (environment.return_pressure() > 30))
+		return TRUE
```
Comment (BongaTheProto) [inline code/modules/mob/mob_movement.dm:221]: You can edit this modularly by rewriting the proc in a `modular_splurt/code/modules/mob/mob_movement.dm` file, adding something like this:

```
/mob/Process_Spacemove(movement_dir = 0)
	var/turf/T = get_turf(src)
	var/datum/gas_mixture/environment = T.return_air()
	if(HAS_TRAIT(src, TRAIT_FLUTTER) && (environment.return_pressure() > 30))
		return TRUE
	. = ..()
```

**Review by BongaTheProto — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob_movement.dm (hunk 1 above)
Comment (ShamanSliph) [reply code/modules/mob/mob_movement.dm:221]: Does anything need to be done to the original, Or do I just need to make the file and put my code there?

Comment (BongaTheProto) [reply code/modules/mob/mob_movement.dm:221]: I suppose delete this edit and put the code in the file

Comment (ShamanSliph) [reply code/modules/mob/mob_movement.dm]: Added the new file and defined it in the DME. So this should be good now.

File: code/__DEFINES/traits.dm [hunk 2]
```diff
@@ -190,6 +190,7 @@
 #define TRAIT_VORACIOUS			"voracious"
 #define TRAIT_SELF_AWARE		"self_aware"
 #define TRAIT_FREERUNNING		"freerunning"
+#define TRAIT_FLUTTER			"flutter"
```
[the remark this answers is not in the document]
Comment (ShamanSliph) [reply code/__DEFINES/traits.dm]: Moved to the right spot. So this should be good.

**Pull request merged**